frontend/src/components/activity/content/LAThematicArea.vue

```diff
@@ -1,46 +1,195 @@
+          <v-skeleton-loader v-else type="sentences" class="mt-2" width="100px" />
+        </template>
+      </template>
+      <template #item="{ item, parent, on, attrs }">
+        <v-list-item three-line class="ec-lta-item" v-bind="attrs" v-on="on">
           <v-list-item-action>
-            <api-checkbox
-              :fieldname="`data.options.${key}.checked`"
-              :uri="contentNode._meta.self"
+            <v-simple-checkbox
+              class="pointer-events-none"
+              :value="parent.hasItem(item)"
+              :color="parent.color"
+              :ripple="false"
             />
           </v-list-item-action>
           <v-list-item-content>
-            <v-list-item-title>{{
-              $tc(`contentNode.laThematicArea.entity.option.${key}.name`)
-            }}</v-list-item-title>
-            <v-list-item-subtitle>{{
-              $tc(`contentNode.laThematicArea.entity.option.${key}.description`)
-            }}</v-list-item-subtitle>
+            <v-list-item-title>{{ item.text }}</v-list-item-title>
+            <v-list-item-subtitle>{{ item.description }}</v-list-item-subtitle>
           </v-list-item-content>
         </v-list-item>
-      </v-list-item-group>
-    </v-list>
+      </template>
+    </e-select>
   </card-content-node>
 </template>
 
 <script>
-import ApiCheckbox from '@/components/form/api/ApiCheckbox.vue'
 import CardContentNode from '@/components/activity/CardContentNode.vue'
 import { contentNodeMixin } from '@/mixins/contentNodeMixin.js'
+import { debounce, isEqual, sortBy } from 'lodash'
+import { serverErrorToString } from '@/helpers/serverError.js'
 
 export default {
   name: 'LAThematicArea',
-  components: { CardContentNode, ApiCheckbox },
+  components: { CardContentNode },
   mixins: [contentNodeMixin],
+  data() {
+    return {
+      localSelection: [],
+      isSaving: false,
+      dirty: false,
+      errorMessages: [],
+      debouncedSave: () => null,
+    }
+  },
+  computed: {
+    dataOptions() {
+      return Object.entries(this.contentNode.data.options)
+    },
+    serverSelection() {
+      return this.dataOptions.filter(([_, option]) => option.checked).map(([key]) => key)
+    },
+    selectionCount() {
+      return this.localSelection.length
+    },
+    items() {
+      return this.dataOptions.map(([key, option]) => ({
+        text: this.$tc(`contentNode.laThematicArea.entity.option.${key}.name`),
+        description: this.$tc(
+          `contentNode.laThematicArea.entity.option.${key}.description`
+        ),
+        value: key,
+        checked: option.checked,
+      }))
+    },
+  },
+  watch: {
+    dataOptions: {
+      async handler(newOptions, oldOptions) {
+        if (isEqual(sortBy(newOptions), sortBy(oldOptions))) {
+          return
+        }
+
+        // copy incoming data if not dirty or if incoming data is the same as local data
+        if (!this.dirty || isEqual(sortBy(newOptions), sortBy(this.localSelection))) {
```

Comment:
`dataOptions` and `localSelection` have a different format, hence the second part of this condition is never true and the dirty flag is never reset to false.

I think you could just do the watcher on `serverSelection` instead (which has the same format as `localSelection`) and then it should work.

Reply:
Good catch. 

Also maybe ApiWrapper could handle these use cases.

Reply:
@usu done

Reply:
@usu done. I've fixed the dirty handling and the also noticed the isSaving was set to false too soon.